### PR TITLE
Group details page refactor

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Updated the User Group Details page to remove DataTables and use Ant Design

## Related issue
#413 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
~~* [ ] User documentation updated for UI or technical changes.~~

### CODE REVIEW

File: src/main/webapp/resources/js/apis/users/groups.js [hunk 1]
```diff
@@ -12,3 +12,59 @@ const BASE_URL = setBaseUrl("/ajax/user-groups");
...
+ * @returns {Promise<AxiosResponse<any>>}
+ */
+export function updateUserGroupDetails({ id, field, value }) {
+  return axios.put(`${BASE_URL}/${id}/update `, {
+    field,
+    value,
+  });
+}
+
+export function getUserGroupRoles() {
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/apis/users/groups.js:40]: Add comments to this function and the ones below

File: src/main/webapp/resources/js/components/roles/UserGroupRole.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React, { useContext, useState } from "react";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/components/roles/UserGroupRole.jsx:1]: File comments

File: src/main/webapp/resources/js/contexts/UserGroupRolesContext.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React, { createContext, useEffect, useState } from "react";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/contexts/UserGroupRolesContext.js:1]: File comments

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupDetailsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useReducer, useRef } from "react";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupDetailsPage.jsx:1]: File comments

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupMembersTable.jsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
+import { UserGroupRole } from "../../../components/roles/UserGroupRole";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupMembersTable.jsx:1]: File comments

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupProjectsTable.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React, { useEffect, useState } from "react";
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupProjectsTable.jsx:1]: File comments

File: src/main/webapp/resources/js/apis/users/groups.js (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/apis/users/groups.js:40]: Updated in ac0e9bf28b8bcf0ed4d214cb35ff57d88fc2dbe3

File: src/main/webapp/resources/js/components/roles/UserGroupRole.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/components/roles/UserGroupRole.jsx:1]: updated in d27433ffe185b74a48f0ae9537484f45d7063c91

File: src/main/webapp/resources/js/contexts/UserGroupRolesContext.js (hunk 3 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/contexts/UserGroupRolesContext.js:1]: Updated in d27433ffe185b74a48f0ae9537484f45d7063c91

File: src/main/webapp/resources/js/components/roles/UserGroupRole.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/components/roles/UserGroupRole.jsx:1]: Updated in 3510322c2098dc93241b530c73e1c9381928fed6

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupDetailsPage.jsx (hunk 4 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupDetailsPage.jsx:1]: Updated in 96f283c7fc96e970e49af9238c991bd38da114d0

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupMembersTable.jsx (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupMembersTable.jsx:1]: Updated in b8cc896b208449c4d489d83294088ac0b4660d51

File: src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupProjectsTable.jsx (hunk 6 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/UserGroupsPage/components/UserGroupProjectsTable.jsx:1]: Updated in 57a0262c0e02335402f91d1bf8cbb06e421b4c7b

**Pull request merged**